compiler/testData/codegen/bytecodeText/boxingOptimization/maxMinByOrNull.kt

```diff
@@ -29,7 +29,7 @@ fun box(): String {
 // 12 ICONST_0
 // 1 IF_ICMPGE
 // 1 IF_ICMPLE
-// 4 LCMP
+// 2 LCMP
```

Comment:
This change will probably break IR / FE 1.0 codegen tests

Reply:
Uh, forgot that it's shared across other tests. I dug into what made this change, and found that only explicit, resolved type is worth being propagated to lhs/rhs. With that change, all the spec/diagnostic changes are gone. (Most of previous changes look correct to me, though.) Only changes about the bb test and IR text tests are remained as-is.